### PR TITLE
Added session token to reduce the auth request data size

### LINKED ISSUE #259: Token received by user needs to be a session token that points to db
We can't persist the original tokens as cookies as they are way too large.

Instead, when a user successfully logs in, we need to establish an app-session for that login, which needs to have a similar expiry as the access token. The id of the session needs to be a key into a db table containing the original token (or the parsed UserInfo since it will not change for that specific session).

When a user makes requests into papiea, he can either use s2s-keys or a session-id. Papiea will check the session-id against the db, see if it needs to renew the token using the refresh token, and return the UserInfo of the user for further processing.

The db will need to be periodically cleaned up to remove expired sessions entries.

When a user logs out, we need to clear that session and remove the db entry for that session. 

An attempt to access papiea with an invalid session needs to be redirected to login.

### PR DESCRIPTION
**Pull request by Emulebest**
Resolves #259

### CODE REVIEW

File: papiea-engine/__tests__/auth_tests/entity_api_auth.test.ts [hunk 1]
```diff
@@ -205,7 +207,7 @@ describe("Entity API auth tests", () => {
         await entityApi.delete(`/${provider.prefix}/${provider.version}/${kind_name}/${entity_metadata.uuid}`);
     });
 
-    test("Get user info", async () => {
+    test.only("Get user info", async () => {
```
Comment (adolgarev) [inline papiea-engine/__tests__/auth_tests/entity_api_auth.test.ts]: only?

File: papiea-engine/package-lock.json [hunk 2]
```diff
@@ -5736,613 +5736,6 @@
         "pluralize": "^7.0.0",
         "swagger-model-validator": "^3.0.6",
         "wait-port": "^0.2.2"
-      },
```
Comment (adolgarev) [inline papiea-engine/package-lock.json]: package.json has not changed, please do not push your package-lock.json

File: papiea-engine/src/auth/oauth2.ts [hunk 3]
```diff
@@ -90,11 +76,8 @@ export function createOAuth2Router(logger: Logger, redirect_uri: string, provide
     }));
 
     router.use('/provider/:prefix/:version/auth/logout', asyncHandler(async (req, res) => {
-        const provider: Provider = await providerDb.get_provider(req.params.prefix, req.params.version);
-        const oauth2 = getOAuth2(provider);
-        const token = oauth2.accessToken.create({ "access_token": req.user.authorization.split(' ')[1] });
         try {
-            await token.revoke('access_token');
```
Comment (adolgarev) [inline papiea-engine/src/auth/oauth2.ts]: still need to revoke a token from idp

File: papiea-engine/src/auth/session_key.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    async createKey(userInfo: UserAuthInfo, token: any, key: string): Promise<SessionKey> {
+        const parsedToken = parseJwt(token.token.access_token)
+        const exp = parsedToken.content.exp
+        const sessionKey: SessionKey = {
+            key: key,
+            expireAt: new Date(exp * 1000),
+            user_info: userInfo,
+            idpToken: token
+        }
+        if (SessionKeyAPI.isExpired(sessionKey)) {
```
Comment (shlomi-nx) [inline papiea-engine/src/auth/session_key.ts]: We should attempt to use the `refresh_token` to extend the `access_token`'s duration

File: papiea-engine/src/auth/session_key.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+export class SessionKeyAPI {
+    private readonly sessionKeyDb: SessionKeyDb
+
+    constructor(sessionKeyDb: SessionKeyDb) {
+        this.sessionKeyDb = sessionKeyDb
+    }
+
+    async createKey(userInfo: UserAuthInfo, token: any, key: string): Promise<SessionKey> {
+        const parsedToken = parseJwt(token.token.access_token)
```
Comment (shlomi-nx) [inline papiea-engine/src/auth/session_key.ts]: Is `access_token` a common oauth2 thing? Or should we add a portion in our auth.yml dsl to capture that?

File: papiea-engine/src/auth/session_key.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    async getKey(key: string): Promise<SessionKey> {
+        const sessionKey = await this.sessionKeyDb.get_key(key)
+        if (!SessionKeyAPI.isExpired(sessionKey)) {
+            return sessionKey
+        } else {
+            throw new Error("Access token expired")
+        }
+    }
+
+    async inActivateKey(key: string) {
```
Comment (shlomi-nx) [inline papiea-engine/src/auth/session_key.ts]: "Inactivate" is a single word verb, it should be `inactivateKey`

File: papiea-engine/src/auth/session_key.ts (hunk 4 above)
Comment (shlomi-nx) [reply papiea-engine/src/auth/session_key.ts]: https://github.com/nutanix/papiea-js/issues/262

File: papiea-engine/src/auth/session_key.ts (hunk 5 above)
Comment (shlomi-nx) [reply papiea-engine/src/auth/session_key.ts]: https://github.com/nutanix/papiea-js/issues/263

Comment (Emulebest) [reply papiea-engine/src/auth/session_key.ts]: It is a standard for OpenID

File: papiea-engine/src/auth/session_key.ts (hunk 6 above)
Comment (Emulebest) [reply papiea-engine/src/auth/session_key.ts]: my bad, will do

**Pull request merged**